src/integrator/cuda/cuda_inc_potential.cu

```diff
@@ -7,16 +7,25 @@ namespace cuda       {
 
 using namespace GauXC::cuda;
 
-template <typename T>
-__global__ void inc_by_submat_combined_kernel( size_t           ntasks,
-                                               XCTaskDevice<T>* device_tasks,
-                                               T*               A,
-                                               size_t           LDA ) {
 
-  const int batch_id = blockIdx.z;
+#define WARP_X 32
+#define WARP_Y 1
+#define UNROLL_FACTOR 4
+#define EFF_UNROLL 4
+#define CUT_X 8
+#define CUT_Y 4
 
-  if( batch_id < ntasks ) {
 
+template <typename T>
+__global__ __launch_bounds__(1024, 1)
```

Comment:
For an unroll factor of 4, there is no register spilling. 

I mainly had it there to prevent launch errors, as the thread-block is large and the register usage is pretty high.

Reply:
Ok, thanks for clarifying